bound/bound_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package bound
+
+import (
+	"testing"
+)
+
+func isIdentical(a, b []Bound) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	for i := 0; i < len(a); i++ {
+		if a[i] != b[i] {
+			return false
+		}
+	}
+
+	return true
+}
+func TestBoundInterSect(t *testing.T) {
+	b0 := Bound{Min: 0.0, Max: 5.0}
+	b1 := Bound{Min: 3.0, Max: 4.0}
+	b2 := Bound{Min: -1.0, Max: 1.0}
+	b3 := Bound{Min: 3.5, Max: 4.5}
+	b4 := Bound{Min: 7.0, Max: 9.0}
+
+	ret := Intersection(b0)
+	expected := []Bound{
+		Bound{Min: 0.0, Max: 5.0},
+	}
+	if !isIdentical(ret, expected) {
+		t.Errorf("Intersection should be %v but got %v", expected, ret)
+	}
+
+	ret = Intersection(b0, b1)
+	expected = []Bound{
+		Bound{Min: 3.0, Max: 4.0},
+	}
+
+	if !isIdentical(ret, expected) {
+		t.Errorf("Intersection should be %v but got %v", expected, ret)
+	}
+
+	ret = Intersection(b0, b1, b2)
+	expected = []Bound{
+		Bound{Min: 0.0, Max: 1.0},
+	}
+
+	if !isIdentical(ret, expected) {
+		t.Errorf("Intersection should be %v but got %v", expected, ret)
+	}
+
+	ret = Intersection(b1, b4)
+	expected = []Bound{}
+
+	if !isIdentical(ret, expected) {
+		t.Errorf("Intersection should be %v but got %v", expected, ret)
+	}
+
+	ret = Intersection(b0, b1, b3)
```

Comment:
@btracey 
Thanks for the fast review.
If we update the signature of this API into `func Intersection(bounds ...Bound) Bound`
What is the expected result of this API?

Reply:
or did you intend `Intersect(Intersect(b0, b1), b3))`?

Reply:
My intended behavior is `Intersect(Intersect(b0, b1), b2))`, but then made easy with the variadic input. This way you can write `Intersection(b0, b1)`, which as you say is the common case, but also could write `Intersection(bounds...)` if you have a list of them.